### DISCUSSION TITLE
Errors with map/apply with pyright "basic"

### DISCUSSION THREAD

**Discussion by PhanAlexandre** [pandas-dev/pandas-stubs#1002] [Q&A]
Hi,
I have some errors with pyright and pandas. I have ask the question there and they send me here.
https://github.com/microsoft/pyright/discussions/8985

Its seems the installing pandas-stubs resolve some of those errors (all but .apply(Path) on one computer but for the other its still showing errors on all line.
Here is some line of code that show this errors

```
import pandas as pd
from pathlib import Path

df = pd.DataFrame({"a":["a","a","b","b","c"]})

df['b'] = df['a'].map(Path) #No overloads for "map" match the provided arguments
df['c'] = df['b'].apply(Path) #No overloads for "map" match the provided arguments
df['b'] = df['a'].map(list) #No overloads for "map" match the provided arguments
df['c'] = df['b'].apply(list)
df['b'] = df['a'].map(set) #No overloads for "map" match the provided arguments
df['c'] = df['b'].apply(set)
```
Is it normal ?

I am on pandas 2.2.2
pylance v2024.9.1
on the computer with less errors pandas-stubs-2.1.4.231227
and the other pandas-stubs-2.2.2.240909

**Comment by twoertwein** [comment]
Pandas-stubs has made the trade-off of allowing only a subset of common types on Series to provide better operator/method support for those types. Whether this trade-off was a good decision is a different question :). Unfortunately, the Python static type system cannot model operators of container types properly (see for example https://github.com/microsoft/pyright/discussions/6549), except if you enumerate all of them which is not maintainable.

I think it might be possible to at least allow the call but return `Series[Unknown/Any]`.

Similar to https://github.com/pandas-dev/pandas-stubs/issues/1001

**Reply by PhanAlexandre** [reply]
Thanks for the answer. However, its still strange that the newest version of pandas-stud show more errors than  pandas-stubs-2.1.4.231227 (which show only 'df['c'] = df['a'].apply(Path)' as an errors)
And that map and apply don't show all the same errors.

I fond a workaround by creating a function likes, def x: return path with typing ->Any.

**Reply by twoertwein** [reply]
Thank you for pointing out that this behavior changed! I will later today/tomorrow look through recent PRs that might have changed that behavior to see whether that was an (un)intended change.

**Reply by twoertwein** [reply]
apply and map used to have some un-annotated parameters so you didn't see errors in older versions.

I think we should add a fallback overload for `Series.apply` and `Series.map`, that accepts `Any` and returns `Series | DataFrame` (or `Any`). This would also fix #1001 @Dr-Irv 

For issues in pandas, please open an issue there.
